app/controllers/concerns/foreman/controller/registration_command.rb

```diff
@@ -0,0 +1,66 @@
+module Foreman::Controller::RegistrationCommand
+  extend ActiveSupport::Concern
+
+  private
+
+  def command
+    args_query = "?#{registration_args.to_query}"
+    "curl #{insecure} -X GET '#{endpoint}#{args_query if args_query != '?'}' #{command_headers} | bash"
+  end
+
+  def registration_args
+    args = registration_params.except(*ignored_query_args)
+
+    args['setup_insights'] = setup_insights.to_s
+    args['setup_remote_execution'] = setup_remote_execution.to_s
+
+    args.delete_if { |_, v| v.blank? }
+        .permit!
+  end
+
+  def insecure
+    registration_params['insecure'] ? '--insecure' : ''
+  end
+
+  def endpoint
+    return global_registration_url if registration_params['smart_proxy_id'].blank?
+
+    proxy = SmartProxy.authorized(:view_smart_proxies).find(registration_params['smart_proxy_id'])
+    "#{proxy.url}/register"
+  end
+
+  def host_config_params
+    organization = Organization.authorized(:view_organizations).find(registration_params['organization_id']) if registration_params['organization_id'].present?
+    location = Location.authorized(:view_locations).find(registration_params['location_id']) if registration_params['location_id'].present?
+    host_group = Hostgroup.authorized(:view_hostgroups).find(registration_params['hostgroup_id']) if registration_params["hostgroup_id"].present?
+    operatingsystem = Operatingsystem.authorized(:view_operatingsystems).find(registration_params['operatingsystem_id']) if registration_params["operatingsystem_id"].present?
+
+    Host.new(organization: organization, location: location, hostgroup: host_group, operatingsystem: operatingsystem).params
+  end
+
+  def setup_insights
+    return if registration_params['setup_insights'].to_s.blank?
```

Comment:
Just curiosity: why `.to_s` cast is needed?

```ruby
registration_params['setup_insights'] = nil
registration_params['setup_insights'].blank? #=> true
registration_params['setup_insights'] = false
registration_params['setup_insights'].blank? #=> true
# but
registration_params['setup_insights'] = false.to_s
registration_params['setup_insights'].blank? #=> false
```

Just a bit hard to follow why exactly in this way and what exactly you want to achieve.

Reply:
Because I want to be sure that `registration_params['setup_...]` will be always `String`, and not `Boolean`.

In typical scenario, `registration_params` will be always `String`, but when you write tests or can the methods from somewhere else you can accidentally set it to `Boolean` , which cause this behavior:
```ruby
registration_params['setup_insights'] = false
return if registration_params['setup_insights'].blank? #=> true
# unreachable code, so it's a bug

registration_params['setup_insights'] = 'false'
return if registration_params['setup_insights'].blank? #=> false
# tada! now you can run logic for setting value of parameter
```
But I agree that the could could be simply:
```
return unless registration_params['setup_insights'].to_s
```